cmake/public/cuda.cmake

```diff
@@ -60,11 +60,15 @@ find_package(CUDAToolkit REQUIRED)
 
 cmake_policy(POP)
 
+# resolve symlinks: GH-108931
+file(REAL_PATH CUDA_INCLUDE_DIRS _inc_cuda)
+file(REAL_PATH CUDAToolkit_INCLUDE_DIR _inc_ctk)
 if(NOT CMAKE_CUDA_COMPILER_VERSION STREQUAL CUDAToolkit_VERSION OR
-    NOT CUDA_INCLUDE_DIRS STREQUAL CUDAToolkit_INCLUDE_DIR)
-  message(FATAL_ERROR "Found two conflicting CUDA installs:\n"
-                      "V${CMAKE_CUDA_COMPILER_VERSION} in '${CUDA_INCLUDE_DIRS}' and\n"
-                      "V${CUDAToolkit_VERSION} in '${CUDAToolkit_INCLUDE_DIR}'")
+   NOT _inc_cuda STREQUAL _inc_ctk)
```

Comment:
Hi! Would you as well consider making this `IN_LIST` instead of `STREQUAL`? I'm encountering a related error when using a splayed CUDA installation (my `CUDAToolkit_INCLUDE_DIR` is a semincolon-separated list of paths to individual CUDA components):

```
  Found two conflicting CUDA installs:

  V11.8.89 in
  '/nix/store/cm0kgn5995f8s6wddp9pgf172cr4280h-cuda_cudart-11.8.89/include'
  and

  V11.8.89 in
  '...;/nix/store/cm0kgn5995f8s6wddp9pgf172cr4280h-cuda_cudart-11.8.89/include;/nix/store/iyzbvj3qlq4b99l1r5507ms9hq5cl1d2-cuda_cupti-11.8.87-dev/include;/nix/store/3ygzadn7c72k5i10qffzainzgmd27dpf-cuda_nvcc-11.8.89-dev/include;/nix/store/5h2lnvvdh29mwyaind8s4rgy5fry6ik9-cuda_nvml_dev-11.8.86-dev/include;/nix/store/lqpdlm7q5a2jy5w2qkc2b2hw4048i3zk-cuda_nvrtc-11.8.89-dev/include;/nix/store/i46q6mggmvcddk9hm0d36mxrbvh4di8c-cuda_nvtx-11.8.86-dev/include;/nix/store/7jx76sywhvhy1viwh00y71r3yyamji9d-libcublas-11.11.3.6-dev/include;/nix/store/x1vbhxmkwg62yfn5kvjj16lr3r4fc6n4-libcufft-10.9.0.58-dev/include;/nix/store/1kfria0ian7hrm8cy1xhpzjaw5zcgw1r-libcurand-10.3.0.86-dev/include;/nix/store/wzqs5agmc5mj89267cl24gyab5s31c7n-libcusolver-11.4.1.48-dev/include;/nix/store/l4lq7k62wms1q9y39jv8pv2qsr4zyn99-libcusparse-11.7.5.86-dev/include;/nix/store/zpjpcd8snwr8znx56kqbllpmg990n1hf-cuda_profiler_api-11.8.86-dev/include'
```